### PR TITLE
added OTP P Sample

### PR DESCRIPTION
**Pull request by ShivKushwah**

### CODE REVIEW

File: OTP-P-Sample/OTPServer.p [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    start state Init {
+	    entry {
+			clientMachine = new CLIENT_OTP_GENERATOR();
+			bankMachine = new BANK_SERVER(clientMachine);
+	    }
+	}
+}
+
+
+fun _CREATECONTAINER() : machine
```
Comment (ShivKushwah) [inline OTP-P-Sample/OTPServer.p]: For some reason, if I comment out this function, my program crashes. I am unable to find references to the usage of this function in the code.

File: OTP-P-Sample/OTPServer.p [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    start state Init {
+        on OTPSecretMsg goto HandleOTPSecret;
+    }
+
+    state HandleOTPSecret {
+	    entry (payload: (machine, int)) {
+	        bankServer = payload.0;
+			OTPSecret = payload.1;
+			send bankServer, OTPSecretReceived;
+			raise (Success);
```
Comment (ankushdesai) [inline OTP-P-Sample/OTPServer.p]: You can delete the success event and replace `raise(Success)` by `goto GenerateOTPCode;`

File: OTP-P-Sample/OTPServer.p [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+     state WaitOTPCode {
+        on OTPCodeMsg goto ValidateOTPCode;
+     }
+
+	 state ValidateOTPCode {
+        entry (payload: int) {
+		  // validate OTP code
+		  if (payload == 123456789) {
+			send clientOtpGenerator, OTPCodeValidated;
+          	raise (Success);
```
Comment (ankushdesai) [inline OTP-P-Sample/OTPServer.p]: here also you can delete the success and done events and replace raise by direct jumps to states using goto.

File: OTP-P-Sample/OTPServer.p [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+machine BANK_SERVER 
+{
+    var clientOtpGenerator: machine;
+	var secret: int;
+
+    // This is the entry point.
+    start state Init {
+        entry (payload:any) {
+          clientOtpGenerator = payload as machine;
+          raise (Success);
```
Comment (ankushdesai) [inline OTP-P-Sample/OTPServer.p]: delete success event and replace 'raise(successs)' with goto nameofstate;

**Review by ankushdesai — CHANGES_REQUESTED** [review]

**Comment by ankushdesai** [comment]
This is great, as next step please take to Sumukh on how to work with Foreign functions and get started on invoking SGX functions with this.

**Pull request merged**